tchannel/tchannel.py

```diff
@@ -240,6 +243,12 @@ def advertise(self, routers, name=None, timeout=None):
             When unable to make our first advertise request to Hyperbahn.
             Subsequent requests may fail but will be ignored.
         """
+        if routers is None and router_file is not None:
+            # should just let the exceptions fly
+            with open(router_file, 'r') as json_data:
+                routers = json.load(json_data)
+                json_data.close()
```

Comment:
@junchaowu It's probably okay to let the `json.load` exception bubble up to the user without us catching it. Any reason to catch it?